src/foolscap/connections/tor.py

```diff
@@ -60,33 +61,30 @@ def hint_to_endpoint(self, hint, reactor):
         # bytes with the hostname when talking to the SOCKS server, so the
         # py2 automatic unicode promotion blows up
         host = host.encode("ascii")
-        ep = txtorcon.TorClientEndpoint(host, portnum,
-                                        socks_hostname=self._socks_hostname,
-                                        socks_port=self._socks_portnum)
+        ep = txtorcon.TorClientEndpoint(host, portnum, socks_endpoint=self._socks_endpoint)
         returnValue( (ep, host) )
 
 
 # note: TorClientEndpoint imports 'reactor' itself, doesn't provide override.
 # This will be fixed in txtorcon 1.0
 
 class _SocksTor(_Common):
-    def __init__(self, hostname=None, portnum=None):
+    def __init__(self, socks_endpoint=None):
         _Common.__init__(self)
         self._connnected = True # no need to call _connect()
-        self._socks_hostname = hostname
-        self._socks_portnum = portnum
-        # portnum=None means to use defaults: 9050, then 9150
+        self._socks_endpoint = socks_endpoint
+        # socks_endpoint=None means to use defaults: TCP to 127.0.0.1 with 9050, then 9150
     def _connect(self, reactor):
         return succeed(None)
 
 def default_socks():
     # TorClientEndpoint knows how to cycle through a built-in set of socks
     # ports, but it doesn't know to set the hostname to localhost
-    return _SocksTor("127.0.0.1")
+    return _SocksTor()
 
-def socks_port(host, portnum):
-    assert isinstance(portnum, int)
-    return _SocksTor(host, portnum)
+def socks(socks_endpoint):
```

Comment:
let's name this function `socks_endpoint()`, to match the `control_endpoint()` below. (it might be necessary to change the argument name to `tor_socks_endpoint`, to avoid a collision with the function name.. that's what I had to do with `control_endpoint()`)